### PR TITLE
feat: add progress and category

### PR DESCRIPTION
**Pull request by Lehren**
Add category and progress model.

### CODE REVIEW

File: src/administrative-sdk/category/category.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * A category for a challenge.
+ * It is used to give a challenge front end attributes such as an icon, a color palette etc.
+ */
+export default class Category {
+  /**
+   * Create a category.
+   *
+   * @param {?string} id - Identifier of this category. If none is given one is generated upon creation in the API.
+   * @param {string} name - Name associated with this category.
+   * @param {string} description - Description associated with this category.
```
Comment (joggienl) [inline src/administrative-sdk/category/category.js]: Should be optional

File: src/administrative-sdk/category/category.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   * @param {string} color - Color associated with this category.
```
Comment (joggienl) [inline src/administrative-sdk/category/category.js]: Should be optional

File: src/administrative-sdk/category/category.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+   * @param {string} image - Image associated with this category.
```
Comment (joggienl) [inline src/administrative-sdk/category/category.js]: Should be optional

File: src/administrative-sdk/category/category.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+   * @param {string} icon - Icon associated with this category.
```
Comment (joggienl) [inline src/administrative-sdk/category/category.js]: Should be optional

File: src/administrative-sdk/category/category.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+   * @param {?Array.<Category>} categories - Other categories this category contains. A category can only contain
+   * either more categories or challenges.
+   * @param {?Array.<SpeechChallenge>} speechChallenges - Challenges this category contains. A category can only contain
+   * either more categories or challenges.
+   * @param {?Progress} progress - Progress values of this category.
```
Comment (joggienl) [inline src/administrative-sdk/category/category.js]: Should be optional

Comment (Lehren) [inline src/administrative-sdk/category/category.js]: fixed

**Pull request merged**